.circleci/config.yml

```diff
@@ -0,0 +1,52 @@
+version: 2
+jobs:
+  test_jdk8_with_coverage:
+    working_directory: ~/flo
+    docker:
+      - image: maven:3.5.4-jdk-8
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-deps-{{ checksum "pom.xml" }}
+      - run: mvn -B dependency:go-offline compile test-compile
+      - save_cache:
+          key: v1-deps-{{ checksum "pom.xml" }}
+          paths:
+            - ~/.m2
+      - run:
+          environment:
+            # Configure datastore emulator heap size
+            JAVA: java -Xmx512m -Xms512m
+            # Maven heap size
+            MAVEN_OPTS: -Xmx256m -Xms256m
+          command: mvn -B verify scala:doc-jar -Pcoverage
+      - run: bash <(curl -s https://codecov.io/bash)
+
+  test_jdk10:
+    working_directory: ~/styx
+    docker:
+      - image: maven:3.5.4-jdk-10-slim
+    steps:
+      - checkout
+      - restore_cache:
+          key: v1-deps-jdk10-{{ checksum "pom.xml" }}
+      - run: mvn -B dependency:go-offline compile test-compile
+      - save_cache:
+          key: v1-deps-jdk10-{{ checksum "pom.xml" }}
+          paths:
+            - ~/.m2
+      - run: mvn -B verify scala:doc-jar
+
+workflows:
+  version: 2
+  build:
+    jobs:
+      - test_jdk8_with_coverage:
+          filters:
+            branches:
+              ignore: gh-pages
+# FIXME
```

Comment:
What's the issue to be fixed here?

Reply:
Too much for this PR. There is another wip PR for that.